src/backend/model_deployments/sagemaker.py

```diff
@@ -73,6 +78,17 @@ def is_available(cls) -> bool:
     def invoke_chat_stream(
         self, chat_request: CohereChatRequest, **kwargs: Any
     ) -> Generator[StreamedChatResponse, None, None]:
+        # TODO: implement metrics correctly
+        start_time = time.perf_counter()
+        metrics_data = MetricsData(
+            endpoint_name="co.chat",
+            method="POST",
+            trace_id=kwargs.pop("trace_id", None),
+            user_id_hash=hash_string(kwargs.pop("user_id", None)),
+            model=chat_request.model,
+            success=True,
+        )
```

Comment:
so we're metering the route on toolkit which calls this and is also metered? (if im understanding correctly?) is this what we want to do, maybe now or later we can just use traces.

same to all the other calls we're doing in the other deployments, etc.